### PR TITLE
Fmi observations

### PR DESCRIPTION
**Pull request by maakuth**
Here's the #8 fix, work in progress. The API client seems to work okay, but I'm yet to integrate it with the chart drawing stuff in the frontend.

### CODE REVIEW

**Comment by vesanieminen** [comment]
Thanks a bunch! I'll take a look at this later.

File: src/main/java/com/vesanieminen/froniusvisualizer/views/NordpoolspotView.java [hunk 1]
```diff
@@ -449,13 +454,26 @@ private DataSeries createEstimateDataSeries(List<FingridLiteResponse> dataSource
...
-        for (SpotHintaResponse response : dataSource) {
-            final var dataSeriesItem = new DataSeriesItem();
-            dataSeriesItem.setX(response.TimeStamp.toInstant().plus(Duration.ofHours(2)));
-            dataSeriesItem.setY(response.Temperature);
-            dataSeries.add(dataSeriesItem);
+
+        if (spotHintadataSource != null) {
+            for (SpotHintaResponse response : spotHintadataSource) {
+                final var dataSeriesItem = new DataSeriesItem();
+                dataSeriesItem.setX(response.TimeStamp.toInstant().plus(Duration.ofHours(2)));
```
Comment (maakuth) [inline src/main/java/com/vesanieminen/froniusvisualizer/views/NordpoolspotView.java:463]: I don't know if this addition of two hours in meant to be a timezone conversion? I think the observations might have the same offset issue.

File: src/main/java/com/vesanieminen/froniusvisualizer/views/NordpoolspotView.java [hunk 2]
```diff
@@ -449,13 +454,26 @@ private DataSeries createEstimateDataSeries(List<FingridLiteResponse> dataSource
...
+                dataSeriesItem.setY(response.Temperature);
+                dataSeries.add(dataSeriesItem);
+            }
+        }
+
+        if (fmiObservations != null) {
+            for (FmiObservation observation : fmiObservations.getObservations()) {
+                final var dataSeriesItem = new DataSeriesItem();
+                var timestamp = FmiService.parseFmiTimestamp(observation.getLocaltime(), observation.getLocaltz());
+                dataSeriesItem.setX(timestamp.toInstant());
```
Comment (maakuth) [inline src/main/java/com/vesanieminen/froniusvisualizer/views/NordpoolspotView.java:473]: Perhaps the same hours should be added here.

**Comment by maakuth** [comment]
I don't have vaadin-charts license so I only see the development environment for a second before it's replaced with the license nag screen. But the data is getting populated into the view, it seems.

File: src/main/java/com/vesanieminen/froniusvisualizer/views/NordpoolspotView.java (hunk 1 above)
Comment (maakuth) [reply src/main/java/com/vesanieminen/froniusvisualizer/views/NordpoolspotView.java:463]: I think that's a correct guess, as currently in liukuri.fi the forecasts start from now-1 hours.

Comment (vesanieminen) [reply src/main/java/com/vesanieminen/froniusvisualizer/views/NordpoolspotView.java:463]: Thanks for the find. This is something I wrote very early on in order to get things working. I'll have to rewrite that timezone conversion in a better way.

**Comment by vesanieminen** [comment]
Thanks for your contribution once more! Sorry it took so long for me to get it added.

**Pull request merged**